server/src/grpc/storage_service/mod.rs

```diff
@@ -587,35 +610,55 @@ fn build_schema_from_metric(schema_config: &SchemaConfig, metric: &WriteMetric)
     )
     .is_nullable(false)
     .build()
-    .context(InvalidColumnSchema {
-        column_name: TSID_COLUMN,
+    .map_err(|e| Box::new(e) as _)
+    .context(ErrWithCause {
+        code: StatusCode::BAD_REQUEST,
+        msg: "invalid timestamp column schema to build",
     })?;
 
     // Use (timestamp, tsid) as primary key.
     let tsid_column_schema =
         column_schema::Builder::new(TSID_COLUMN.to_string(), DatumKind::UInt64)
             .is_nullable(false)
             .build()
-            .context(InvalidColumnSchema {
-                column_name: TSID_COLUMN,
+            .map_err(|e| Box::new(e) as _)
+            .context(ErrWithCause {
+                code: StatusCode::BAD_REQUEST,
+                msg: "invalid tsid column schema to build",
             })?;
 
     schema_builder = schema_builder
         .enable_tsid_primary_key(true)
         .add_key_column(timestamp_column_schema)
-        .with_context(|| BuildTableSchema { metric: table_name })?
+        .map_err(|e| Box::new(e) as _)
+        .context(ErrWithCause {
+            code: StatusCode::BAD_REQUEST,
+            msg: "invalid timestamp column to add",
+        })?
         .add_key_column(tsid_column_schema)
-        .with_context(|| BuildTableSchema { metric: table_name })?;
+        .map_err(|e| Box::new(e) as _)
+        .context(ErrWithCause {
+            code: StatusCode::BAD_REQUEST,
+            msg: "invalid tsid column to add",
+        })?;
 
     for col in name_column_map.into_values() {
         schema_builder = schema_builder
             .add_normal_column(col)
-            .with_context(|| BuildTableSchema { metric: table_name })?;
+            .map_err(|e| Box::new(e) as _)
+            .context(ErrWithCause {
+                code: StatusCode::BAD_REQUEST,
+                msg: "invalid normal column to add",
+            })?;
     }
 
-    schema_builder.build().context(BuildTableSchema {
-        metric: &metric.metric,
-    })
+    schema_builder
+        .build()
+        .map_err(|e| Box::new(e) as _)
+        .context(ErrWithCause {
+            code: StatusCode::BAD_REQUEST,
+            msg: "invalid schema to build",
+        })
```

Comment:
invalid -> Invalid

Reply:
No need to make the first word with capital letter.